src/rust/src/note_encryption.rs

```diff
@@ -0,0 +1,94 @@
+use group::GroupEncoding;
+use zcash_note_encryption::{Domain, EphemeralKeyBytes, ShieldedOutput, ENC_CIPHERTEXT_SIZE};
+use zcash_primitives::{
+    consensus::BlockHeight,
+    memo::MemoBytes,
+    sapling::{
+        self,
+        note_encryption::{PreparedIncomingViewingKey, SaplingDomain},
+        SaplingIvk,
+    },
+};
+
+use crate::{params::Network, wallet_scanner::ffi::SaplingShieldedOutput};
+
+pub(crate) fn try_sapling_note_decryption(
+    network: &Network,
+    height: u32,
+    raw_ivk: &[u8; 32],
+    output: SaplingShieldedOutput,
+) -> Result<Box<DecryptedSaplingOutput>, &'static str> {
+    let ivk = parse_and_prepare_sapling_ivk(raw_ivk)
+        .ok_or("Invalid Sapling ivk passed to wallet::try_sapling_note_decryption()")?;
```

Comment:
```suggestion
        .ok_or("Invalid Sapling ivk passed to note_encryption::try_sapling_note_decryption()")?;
```

Reply:
The error message here is from the perspective of C++, where this function appears in the `wallet` namespace.